<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CDialogAddRule</class>
 <widget class="QDialog" name="DialogAddRule">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>345</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Security Rule</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../Resource.qrc">
    <normaloff>:/Resource/Security/SecurityRule.png</normaloff>:/Resource/Security/SecurityRule.png</iconset>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayoutDialogAddRule">
   <property name="margin">
    <number>0</number>
   </property>
   <property name="spacing">
    <number>0</number>
   </property>
   <item row="0" column="0">
    <widget class="QFrame" name="frameBody">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayoutFrameBody">
      <property name="verticalSpacing">
       <number>2</number>
      </property>
      <property name="margin">
       <number>6</number>
      </property>
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBoxRule">
        <property name="title">
         <string>Rule</string>
        </property>
        <layout class="QGridLayout" name="gridLayoutGroupBoxRule">
         <item row="0" column="0">
          <widget class="QLabel" name="labelRuleType">
           <property name="text">
            <string>Type:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="comboBoxRuleType">
           <item>
            <property name="text">
             <string>IP Address</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>IP Address Range</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Country</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>File Filter</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Content Filter</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Regular Expression</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>User Agent</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelComment">
           <property name="text">
            <string>Comment:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="lineEditComment"/>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QStackedWidget" name="stackedWidgetType">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="pageIPAddress">
         <layout class="QGridLayout" name="gridLayoutIPAddress">
          <property name="margin">
           <number>0</number>
          </property>
          <property name="spacing">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QGroupBox" name="groupBoxIPAddress">
            <property name="title">
             <string>IP Address</string>
            </property>
            <layout class="QGridLayout" name="gridLayoutIPAddress_1">
             <item row="0" column="0">
              <spacer name="verticalSpacer_2">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="labelIPAdress">
               <property name="text">
                <string>IP Address:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="lineEditIP">
               <property name="text">
                <string>0.0.0.0</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0" colspan="2">
              <widget class="QLabel" name="labelExampleIP">
               <property name="text">
                <string>Examples: 192.168.1.1, 2001:db8:0:8d3:0:8a2e:70:7344</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <spacer name="verticalSpacer">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageIPRange">
         <layout class="QGridLayout" name="gridLayoutIPRange">
          <property name="margin">
           <number>0</number>
          </property>
          <property name="spacing">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QGroupBox" name="groupBoxIPRange">
            <property name="title">
             <string>IP Address Range</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_9">
             <property name="margin">
              <number>4</number>
             </property>
             <property name="spacing">
              <number>4</number>
             </property>
             <item row="0" column="0" colspan="6">
              <layout class="QGridLayout" name="gridLayoutIPRange_1">
               <property name="spacing">
                <number>2</number>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="labelStartIP">
                 <property name="text">
                  <string>Start IP Address:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSpinBox" name="spinBoxStartIP1">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QSpinBox" name="spinBoxStartIP2">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                </widget>
               </item>
               <item row="0" column="3">
                <widget class="QSpinBox" name="spinBoxStartIP3">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                </widget>
               </item>
               <item row="0" column="4">
                <widget class="QSpinBox" name="spinBoxStartIP4">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="labelEndIP">
                 <property name="text">
                  <string>End IP Address:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSpinBox" name="spinBoxEndIP1">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="2">
                <widget class="QSpinBox" name="spinBoxEndIP2">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="3">
                <widget class="QSpinBox" name="spinBoxEndIP3">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="4">
                <widget class="QSpinBox" name="spinBoxEndIP4">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="labelRangeNetmask">
                 <property name="text">
                  <string>Netmask:</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QSpinBox" name="spinBoxRangeNetmask1">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                 <property name="value">
                  <number>255</number>
                 </property>
                </widget>
               </item>
               <item row="2" column="2">
                <widget class="QSpinBox" name="spinBoxRangeNetmask2">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                 <property name="value">
                  <number>255</number>
                 </property>
                </widget>
               </item>
               <item row="2" column="3">
                <widget class="QSpinBox" name="spinBoxRangeNetmask3">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                 <property name="value">
                  <number>255</number>
                 </property>
                </widget>
               </item>
               <item row="2" column="4">
                <widget class="QSpinBox" name="spinBoxRangeNetmask4">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                 <property name="value">
                  <number>255</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageCountry">
         <layout class="QGridLayout" name="gridLayoutCountry">
          <property name="margin">
           <number>0</number>
          </property>
          <property name="spacing">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QGroupBox" name="groupBoxCountry">
            <property name="title">
             <string>Country</string>
            </property>
            <layout class="QGridLayout" name="gridLayoutCountry_1">
             <item row="0" column="0">
              <spacer name="verticalSpacerCountry_0">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="labelCountryCode">
               <property name="text">
                <string>Two-letter country code:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="lineEditCountry">
               <property name="text">
                <string>GB</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0" colspan="2">
              <widget class="QLabel" name="labelExampleCountry">
               <property name="text">
                <string>Examples: GB, de, fr, AU, ru,...</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <spacer name="verticalSpacerCountry_1">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageFile">
         <layout class="QGridLayout" name="gridLayoutFile">
          <property name="margin">
           <number>0</number>
          </property>
          <property name="spacing">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QGroupBox" name="groupBoxFile">
            <property name="title">
             <string>File Filter</string>
            </property>
            <layout class="QGridLayout" name="gridLayoutFile_1">
             <property name="spacing">
              <number>2</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="labelURI">
               <property name="text">
                <string>URI:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="lineEditURI"/>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageContent">
         <layout class="QGridLayout" name="gridLayoutContent">
          <property name="margin">
           <number>0</number>
          </property>
          <property name="spacing">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QGroupBox" name="groupBoxContent">
            <property name="title">
             <string>Content Filter</string>
            </property>
            <layout class="QGridLayout" name="gridLayoutContent_1">
             <item row="0" column="0">
              <spacer name="verticalSpacerContent">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="1" column="0" colspan="2">
              <widget class="QLineEdit" name="lineEditContent"/>
             </item>
             <item row="2" column="0">
              <widget class="QRadioButton" name="radioButtonMatchAny">
               <property name="text">
                <string>Match Any</string>
               </property>
               <property name="checked">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QRadioButton" name="radioButtonMatchAll">
               <property name="text">
                <string>Match All</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <spacer name="verticalSpacerContent_1">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageRegularExpression">
         <layout class="QGridLayout" name="gridLayoutRegEx">
          <property name="margin">
           <number>0</number>
          </property>
          <property name="spacing">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>Regular Expression</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QLineEdit" name="lineEditRegularExpression"/>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageUserAgent">
         <layout class="QGridLayout" name="gridLayoutUserAgent">
          <property name="margin">
           <number>0</number>
          </property>
          <property name="spacing">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QGroupBox" name="groupBox_2">
            <property name="title">
             <string>User Agent</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QLineEdit" name="lineEditUserAgent"/>
             </item>
             <item>
              <widget class="QCheckBox" name="checkBoxUserAgent">
               <property name="text">
                <string>Regular Expression</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QGroupBox" name="groupBoxOptions">
        <property name="title">
         <string>Options</string>
        </property>
        <layout class="QGridLayout" name="gridLayoutGroupBoxOptions">
         <property name="margin">
          <number>0</number>
         </property>
         <property name="spacing">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="labelAction">
           <property name="text">
            <string>Action:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1" colspan="6">
          <widget class="QComboBox" name="comboBoxAction">
           <property name="currentIndex">
            <number>2</number>
           </property>
           <property name="maxVisibleItems">
            <number>3</number>
           </property>
           <item>
            <property name="text">
             <string>None</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Accept</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Deny</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelExpire">
           <property name="text">
            <string>Expire:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1" colspan="6">
          <widget class="QComboBox" name="comboBoxExpire">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <property name="maxVisibleItems">
            <number>3</number>
           </property>
           <item>
            <property name="text">
             <string>Never (Indefinite)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>At End Of Session</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>After:</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="lineEditDays">
           <property name="enabled">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QLabel" name="labelDays">
           <property name="text">
            <string>Day(s)</string>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QLineEdit" name="lineEditHours">
           <property name="enabled">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="4">
          <widget class="QLabel" name="labelHours">
           <property name="text">
            <string>Hour(s)</string>
           </property>
          </widget>
         </item>
         <item row="2" column="5">
          <widget class="QLineEdit" name="lineEditMinutes">
           <property name="enabled">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="6">
          <widget class="QLabel" name="labelMinutes">
           <property name="text">
            <string>Minute(s)</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <layout class="QGridLayout" name="gridLayoutFooter">
     <item row="0" column="0">
      <spacer name="horizontalSpacerFooter">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="pushButtonOK">
       <property name="text">
        <string>OK</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="pushButtonCancel">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../Resource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>comboBoxRuleType</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackedWidgetType</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>343</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
